ctms/crud.py

```diff
@@ -267,6 +268,86 @@ def get_contacts_by_any_id(
     return data
 
 
+def get_pending_acoustic_records(
+    db: Session,
+    end_time: datetime,
+):
+    """Get all the pending records up to a certain date."""
+    pending_records: List[PendingAcousticRecord] = (
+        db.query(PendingAcousticRecord)
+        .filter(
+            PendingAcousticRecord.update_timestamp < end_time,
+        )
+        .order_by(asc(Email.update_timestamp), asc(Email.email_id))
+        .all()
+    )
+
+    return pending_records
+
+
+def get_pending_records_as_contacts(
+    pending_records: List[PendingAcousticRecord],
+) -> List[ContactSchema]:
+    return [
+        ContactSchema.parse_obj(
+            {
+                "amo": record.email.amo,
+                "email": record.email,
+                "fxa": record.email.fxa,
+                "mofo": record.email.mofo,
+                "newsletters": record.email.newsletters,
+                "vpn_waitlist": record.email.vpn_waitlist,
+            }
+        )
+        for record in pending_records
+    ]
+
+
+def create_pending_acoustic_record(
+    db: Session,
+    email_id: UUID4,
+    pending_record: Optional[PendingAcousticRecordInSchema],
+) -> None:
+    kwargs = {}
+    if pending_record is not None:
+        kwargs = pending_record.dict()
+
+    db_pending_record = PendingAcousticRecord(email_id=email_id, **kwargs)
+    db.add(db_pending_record)
+
+
+def update_pending_acoustic_record(
+    db: Session,
+    email_id: UUID4,
+    pending_record: PendingAcousticRecordInSchema,
+) -> None:
+    updated_pending_record = PendingAcousticRecordInSchema(**pending_record.dict())
+    stmt = insert(PendingAcousticRecord).values(
+        email_id=email_id, **updated_pending_record.dict()
+    )
+    stmt = stmt.on_conflict_do_update(
+        index_elements=[PendingAcousticRecord.email_id],
+        set_=updated_pending_record.dict(),
+    )
+    db.execute(stmt)
+
+
+def delete_pending_record(db: Session, email_id: UUID4):
```

Comment:
This looks like it would delete all "pending" entries for one email_id. But I recall us having a double key where it's email_id AND some other unique-ifier. Am I misremembering? Do you even know what I'm talking about?

Reply:
Maybe the [id](https://github.com/mozilla-it/ctms-api/pull/181/files#diff-4aa7c2a69c18e1f8b33b7cb5f54efdea916ea029661c212d3b1e4de349c91a24R195) could be used, it should be autogenerated; that or we allow a timestamp as well for this deletion. 

Also, I think this depends on how the pull is implemented...? If the pull of records from the Table contains multiple of the same email_id, wouldn't we likely pull the most up to date record when we get the [records as contacts](https://github.com/mozilla-it/ctms-api/pull/181/files#diff-45799bd240716e4cd04a9c1aeb24ad5e995dd9587db55a53d9eb85aa182d3533R288)? 

I think the edge case here is when an update happens, after we fetched the contacts, before we deleted from the table.

Which can be solved with using the same timestamp required for pulling/gets, or to provide the sequence id's (if that's possible, I can test further...)